Documentation/proposals/202305-scrapeclasses.md

```diff
@@ -0,0 +1,235 @@
+of the `Prometheus` spec. An alternative is outlined later, of factoring the class definitions into a separate resource.
+
+Define a scrape classes for use by the monitors, one class may be designated as the default class.
+
+When a resource defines several default scrape classes, it should fail the reconciliation.
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: Prometheus
+spec:
+  scrapeClasses:
+    - name: istio-mtls
+      default: true
+      tlsConfig:
+        caFile: "/etc/istio-certs/root-cert.pem"
+        certFile: "/etc/istio-certs/cert-chain.pem"
+        keyFile: "/etc/istio-certs/key.pem"
+        insecureSkipVerify: true
+
+  # mount the certs from the istio sidecar (shown here for illustration purposes)
+  volumeMounts:
+    - name: istio-certs
+      mountPath: "/etc/istio-certs/"
+  volumes:
+  - emptyDir:
+      medium: Memory
+    name: istio-certs
+```
+
+Any object references in the scrape class definition are assumed to refer to objects in the namespace of the `Prometheus` object.
+
+### PodMonitor Resource
+
+Allow the user to select a scrape class which applies to all endpoints.
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+spec:
+  scrapeClassName: istio-mtls
+  podMetricsEndpoints:
+  - port: http
+    path: /metrics
+```
+
+If the `Monitor` resource has a `tlsConfig` field defined, the Operator will use a merge strategy to combine the `tlsConfig` fields from the PodMonitor object with the `tlsConfig` fields of the scrape class, the `tlsConfig` fields in the `PodMonitor` resource take precedence.
+
+If the resource specifies a scrape class name that isn't defined in the Prometheus/PrometheusAgent object, then the scrape resource is dropped by the operator.
+
+If the resource doesn't specify a scrape class name and the Prometheus/PrometheusAgent object defines a default scrape class, the operator will act as if the PodMonitor resource specified the default scrape class name.
+
+### Probe Resource
+
+Allow the user to select a scrape class for the prober service.
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: Probe
+spec:
+  scrapeClassName: istio-mtls
+```
+
+### ServiceMonitor Resource
+
+Allow the user to select a scrape class for each endpoint.
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+spec:
+  scrapeClassName: istio-mtls
+  endpoints:
+  - port: http
+    path: /metrics
+```
+
+### ScrapeConfig
+
+Allow the user to select a scrape class for the generic scrape configuration.
+
+```yaml
+apiVersion: monitoring.coreos.com/v1alpha1
+kind: ScrapeConfig
+metadata:
+  name: scrape-config
+spec:
+  scrapeClassName: istio-mtls
+  staticConfigs:
+    [...]
+  httpSDConfig:
+    [...]
+  fileSDConfig:
+    [...]
+```
+
+If the monitor resource specifies a scrape class name that isn't defined in the Prometheus/PrometheusAgent object, then the scrape resource is rejected by the operator.
```

Comment:
(nit) it repeats a bit what is written  for PodMonitor. Maybe reorganize like this?

````
### Monitoring resources

If the monitor resource specifies a scrape class name ...

#### PodMonitor

< insert YAML example >

### Probe

Allow the user to select a scrape class for the prober service.

< insert YAML example >

...
````